geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ConverterIngest.scala

```diff
@@ -0,0 +1,113 @@
+/***********************************************************************
+* Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0 which
+* accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+package org.locationtech.geomesa.tools.ingest
+
+import java.io.File
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.{Executors, TimeUnit}
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.commons.io.IOUtils
+import org.geotools.data.{DataStoreFinder, DataUtilities, Transaction}
+import org.geotools.factory.Hints
+import org.geotools.filter.identity.FeatureIdImpl
+import org.locationtech.geomesa.convert.SimpleFeatureConverters
+import org.locationtech.geomesa.utils.classpath.PathUtils
+import org.opengis.feature.simple.SimpleFeatureType
+
+import scala.collection.JavaConversions._
+
+class ConverterIngest(dsParams: Map[String, String],
+                      sft: SimpleFeatureType,
+                      converterConfig: Config,
+                      inputs: Seq[String],
+                      numLocalThreads: Int)
+    extends Runnable with LazyLogging {
+
+  val ds = DataStoreFinder.getDataStore(dsParams)
+
+  override def run(): Unit = {
+    // create schema for the feature prior to Ingest job
+    logger.info(s"Creating schema ${sft.getTypeName}")
+    ds.createSchema(sft)
+
+    if (inputs.head.toLowerCase.startsWith("hdfs://")) {
+      logger.info("Running ingestion in distributed mode")
+      runDistributed()
+    } else {
+      logger.info("Running ingestion in local mode")
+      runLocal()
+    }
+  }
+
+  private def runLocal(): Unit = {
+    class ThreadSafeCounter extends org.locationtech.geomesa.convert.Transformers.Counter {
+      private val (s, f, c) = (new AtomicLong(0), new AtomicLong(0), new AtomicLong(0))
+
+      override def incSuccess(i: Long): Unit   = s.getAndAdd(i)
+      override def incFailure(i: Long): Unit   = f.getAndAdd(i)
+      override def incLineCount(i: Long): Unit = c.getAndAdd(i)
+      override def setLineCount(i: Long): Unit = c.set(i)
+      override def getFailure: Long            = f.get()
+      override def getLineCount: Long          = c.get()
+      override def getSuccess: Long            = s.get()
+    }
+    val counter = new ThreadSafeCounter
+
+    class LocalIngestWorker(file: File) extends Runnable {
+
+      override def run(): Unit = {}
+        val fw        = ds.getFeatureWriterAppend(sft.getTypeName, Transaction.AUTO_COMMIT)
+        val converter = SimpleFeatureConverters.build(sft, converterConfig)
+        val ec        = converter.createEvaluationContext(Map("inputFilePath" -> file.getAbsolutePath), counter)
```

Comment:
probably each worker should have a separate counter to keep track of it's own line number. Maybe you can have a closure over atomicLongs for success and failure but keep line number separate.